modules/platform/nuxeo-automation/nuxeo-automation-features/src/test/java/org/nuxeo/ecm/automation/core/operations/coldstorage/AbstractTestColdStorageOperation.java

```diff
@@ -66,18 +68,52 @@ protected void moveContentToColdStorage(CoreSession session, DocumentModel docum
             context.setInput(documentModel);
             DocumentModel updatedDocModel = (DocumentModel) automationService.run(context, MoveToColdStorage.ID);
+            checkMoveContent(List.of(documentModel), List.of(updatedDocModel));
+        }
+    }
+
+    protected void moveContentToColdStorage(CoreSession session, List<DocumentModel> documents)
+            throws OperationException, IOException {
+        List<String> documentIds = documents.stream().map(DocumentModel::getId).collect(Collectors.toList());
+        try (OperationContext context = new OperationContext(session)) {
+            context.setInput(documents);
+            checkMoveContent(documents, (DocumentModelList) automationService.run(context, MoveToColdStorage.ID));
+        }
+    }
+
+    protected void checkMoveContent(List<DocumentModel> expectedDocs, List<DocumentModel> foundedDocs)
+            throws IOException {
+        assertEquals(expectedDocs.size(), foundedDocs.size());
+        List<String> expectedDocIds = expectedDocs.stream().map(DocumentModel::getId).collect(Collectors.toList());
+        for (DocumentModel updatedDocModel : foundedDocs) {
             Blob fileContent = (Blob) updatedDocModel.getPropertyValue(ColdStorageHelper.FILE_CONTENT_PROPERTY);
             Blob coldStorageContent = (Blob) updatedDocModel.getPropertyValue(
                     ColdStorageHelper.COLD_STORAGE_CONTENT_PROPERTY);
-            assertEquals(documentModel.getRef(), updatedDocModel.getRef());
-            assertTrue(updatedDocModel.hasFacet(FacetNames.COLD_STORAGE));
-            assertEquals(DummyThumbnailFactory.DUMMY_THUMBNAIL_CONTENT, fileContent.getString());
-            assertEquals(FILE_CONTENT, coldStorageContent.getString());
+
+            // check document
+            assertTrue(String.format("Doc: %s must one of %s", updatedDocModel, expectedDocs),
+                    expectedDocIds.contains(updatedDocModel.getId()));
+            assertTrue(String.format("Doc: %s must have the %s Facet", updatedDocModel, FacetNames.COLD_STORAGE),
+                    updatedDocModel.hasFacet(FacetNames.COLD_STORAGE));
+
+            // check blobs
+            assertEquals(String.format("Doc: %s must have %s in 'file:content'", //
+                    updatedDocModel, DummyThumbnailFactory.DUMMY_THUMBNAIL_CONTENT), //
+                    DummyThumbnailFactory.DUMMY_THUMBNAIL_CONTENT, fileContent.getString());
+            assertEquals(String.format("Doc: %s must have %s in 'coldstorage:coldContent'", //
+                    updatedDocModel, FILE_CONTENT), //
+                    FILE_CONTENT, coldStorageContent.getString());
         }
+
     }
 
     protected DocumentModel createFileDocument(CoreSession session, boolean withBlobContent, ACE... aces) {
-        DocumentModel documentModel = session.createDocumentModel("/", "MyFile", "File");
+        return createFileDocument(session, withBlobContent, "MyFile", aces);
+    }
```

Comment:
What about just removing this method used 2 times, and give a doc name to both calls?

Reply:
you mean just give the same name to all callers ? or mainly your idea is to keep only the method that take a name and i will do a little refactoring :)

Reply:
I remove the new method and i use the same name and the listener responsible of renaming will take it into account.